File: go/etl/etl.go
package etl

import "strings"

const testVersion = 1

func Transform(oldScores map[int][]string) map[string]int {
	result := map[string]int{}
	for score, chars := range oldScores {
		for _, char := range chars {
			result[strings.ToLower(char)] = score
		}
	}
	return result
}

File: haskell/etl/src/ETL.hs
module ETL (transform) where

import Data.Map (Map, toList, fromList)
import Data.Char (toLower)

transform :: Map a String -> Map Char a
transform legacyData = Data.Map.fromList $ foldr f [] (Data.Map.toList legacyData)
  where f (k, x:xs) memo = f (k, xs) ((toLower x, k):memo)
        f (_, []) memo = memo
